exec/java-exec/src/test/java/org/apache/drill/test/TestGracefulShutdown.java

```diff
@@ -205,6 +212,47 @@ public void testDrillbitWithSamePortContainsShutdownThread() throws Exception {
     }
   }
 
+  @Test // DRILL-7056
+  public void testDrillbitTempDir() throws Exception {
+    File originalDrillbitTempDir = null;
+    ClusterFixtureBuilder fixtureBuilder = ClusterFixture.bareBuilder(dirTestWatcher).withLocalZk()
+        .configProperty(ExecConstants.ALLOW_LOOPBACK_ADDRESS_BINDING, true)
+        .configProperty(ExecConstants.INITIAL_USER_PORT, QueryTestUtil.getFreePortNumber(31170, 300))
+        .configProperty(ExecConstants.INITIAL_BIT_PORT, QueryTestUtil.getFreePortNumber(31180, 300));
+
+    try (ClusterFixture fixture = fixtureBuilder.build();
+        Drillbit twinDrillbitOnSamePort = new Drillbit(fixture.config(),
+            fixtureBuilder.configBuilder().getDefinitions(), fixture.serviceSet())) {
+      // Assert preconditions :
+      //      1. First drillbit instance should be started normally
+      //      2. Second instance startup should fail, because ports are occupied by the first one
+      Drillbit originalDrillbit = fixture.drillbit();
+      assertNotNull("First drillbit instance should be initialized", originalDrillbit);
+      originalDrillbitTempDir = getWebServerTempDirPath(originalDrillbit);
+      assertTrue("First drillbit instance should have a temporary Javascript dir initialized", originalDrillbitTempDir.exists());
+      try {
+        twinDrillbitOnSamePort.run();
+        fail("Invocation of 'twinDrillbitOnSamePort.run()' should throw UserException");
+      } catch (UserException e) {
+        assertNull("Second drillbit instance should NOT have a temporary Javascript dir", getWebServerTempDirPath(twinDrillbitOnSamePort));
```

Comment:
`webServerHandle.getOrCreateTmpJavaScriptDir()` method actually creates or returns created temp dir, is it required to call it on the non-started drillbit?

Reply:
I test by calling the method because the non-started drillbit is still instantiated, but failed to come up. 
The method will not create a temp dir if the startup failed. I thought it was a valid test.

Reply:
Actually, the check you have added in `getOrCreateTmpJavaScriptDir()` doesn't guarantee that drillbit is started. Context may be non-nullable since exception may happen later, but drill is not started. Please revert that check. Also, I think this method should not be called on closed drillbit, so it should be removed from the test.